lib/Macros/Sources/SwiftMacros/SwiftifyImportMacro.swift

```diff
@@ -920,13 +1044,17 @@ public struct SwiftifyImportMacro: PeerMacro {
         arguments = arguments.dropLast()
       }
       var nonescapingPointers = Set<Int>()
+      var lifetimeDependencies : [SwiftifyExpr: [LifetimeDependence]] = [:]
       var parsedArgs = try arguments.compactMap {
-        try parseMacroParam($0, funcDecl.signature, nonescapingPointers: &nonescapingPointers)
+        try parseMacroParam($0, funcDecl.signature, nonescapingPointers: &nonescapingPointers,
+          lifetimeDependencies: &lifetimeDependencies)
       }
-      parsedArgs.append(contentsOf: try parseCxxSpanParams(funcDecl.signature, typeMappings))
+      parsedArgs.append(contentsOf: try parseCxxSpansInSignature(funcDecl.signature, typeMappings))
       setNonescapingPointers(&parsedArgs, nonescapingPointers)
+      setLifetimeDependencies(&parsedArgs, lifetimeDependencies)
+      // We only transform non-escaping spans.
       parsedArgs = parsedArgs.filter {
-        !($0 is CxxSpan) || ($0 as! CxxSpan).nonescaping
+        !($0 is CxxSpan) || ($0 as! CxxSpan).nonescaping || ($0 as! CxxSpan).pointerIndex == .return
```

Comment:
How about something like this?
```suggestion
        if let cxxSpanArg = $0.as(CxxSpan) {
          cxxSpanArg.nonescaping || cxxSpanArg.pointerIndex == .return
        } else {
          true
        }
```